ext/spl/spl_observer.c

```diff
@@ -189,6 +247,9 @@ void spl_object_storage_addall(spl_SplObjectStorage *intern, spl_SplObjectStorag
 	intern->index = 0;
 } /* }}} */
 
+#define SPL_OBJECT_STORAGE_CLASS_HAS_OVERRIDE(class_type, method) \
+	(((zend_function *)zend_hash_str_find_ptr(&(class_type)->function_table, (method), sizeof((method)) - 1))->common.scope != spl_ce_SplObjectStorage)
```

Comment:
I added interned strings for these in https://github.com/php/php-src/commit/f0dd79a7e415d8b1d4a3868a27975d6578c87961, you can use them here.

Reply:
I see - the hash table would be much faster and is short circuited if the `zend_string *` pointer was the same.

There were a few more in spl but those should be a separate PR

```
» ag '"offsetget"'     
Zend/zend_string.h
573:    _(ZEND_STR_OFFSETGET,              "offsetget") \

ext/spl/spl_dllist.c
370:            intern->fptr_offset_get = zend_hash_str_find_ptr(&class_type->function_table, "offsetget", sizeof("offsetget") - 1);

ext/spl/spl_fixedarray.c
271:            methods.fptr_offset_get = zend_hash_str_find_ptr(&class_type->function_table, "offsetget", sizeof("offsetget") - 1);
391:            zend_call_method_with_1_params(object, intern->std.ce, &intern->methods->fptr_offset_get, "offsetGet", rv, offset);

ext/spl/spl_array.c
206:            intern->fptr_offset_get = zend_hash_str_find_ptr(&class_type->function_table, "offsetget", sizeof("offsetget") - 1);
425:                    zend_call_method_with_1_params(object, object->ce, &intern->fptr_offset_get, "offsetGet", rv, offset);
```